### PR TITLE
epub_loader

### PR DESCRIPTION
**Pull request by ffreemt**
load epub to a list of Documents with metadata = {"sourc…e": file_path, "ch.": <chapter/section title}

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -18,3 +18,6 @@ click
 
 # Excel File Manipulation
```
Comment (SharePuto) [inline requirements.txt:19]: Déjate de mamadas ya ala verga

File: epub_loader.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            ch_titles = epub2txt.content_titles
+
+        except Exception as exc:
+            logger.error(exc)
+            raise
+
+        docs = []
+        for title, text in zip(ch_titles, texts):
+            metadata = {"source": self.file_path, "ch.": title}
+            docs.append(Document(page_content=text, metadata=metadata))
```
Comment (SharePuto) [inline epub_loader.py:36]: /Append
/details

File: epub_loader.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        return docs
```
Comment (SharePuto) [inline epub_loader.py:38]: - - [ ] - _[@]()__****_

**Review by SharePuto — CHANGES_REQUESTED** [review]

File: epub_loader.py (hunk 3 above)
Comment (SharePuto) [reply epub_loader.py:38]: Amime la pelas verga tu solo ven

File: requirements.txt [hunk 4]
```diff
@@ -18,3 +18,6 @@ click
...
 openpyxl
+
+# for EpubLoader
+epub2txt
```
Comment (sime2408) [inline requirements.txt:23]: Why not use UnstructuredEPubLoader from langchain?

Comment (ffreemt) [reply requirements.txt:23]: Right good suggestion. But somehow UnstructuredEPubLoader('....epub') always hangs in my setup. 

I have  Windows 10, python 3.10, the latest langchain. 

I tried all sorts of epub files，a non-existent epub file (straight from the langchain doc `loader = UnstructuredEPubLoader("winter-sports.epub"`). It always hangs indefinitely. Even Ctrl-C doesn't work.

My EpubLoader will always happily load or raise an exception for a non-existent file.

Comment (sime2408) [reply requirements.txt:23]: @ffreemt In [my project](https://github.com/sime2408/scrapalot-chat/blob/main/scripts/app_utils.py#L52), it is working just fine, but not in the way that I provide name of the file through constructor 👀

Comment (ffreemt) [reply requirements.txt:23]: OK I tested UnstructuredEPubLoader in Linux. It runs OK.

However, UnstructuredEPubLoader parses an epub file to only one single doc hence the metadata is just 
```
[{'source': 'docs\\test.epub'}]
``` 
while EpubLoader'metadata parses an epub file to several docs with chapter info when available. The metadata is something like:
```
[{'source': 'docs\\test.epub', 'ch.': 'NA'},
 {'source': 'docs\\test.epub', 'ch.': 'NA'},
 {'source': 'docs\\test.epub', 'ch.': '1'},
 {'source': 'docs\\test.epub', 'ch.': '1*'},
 {'source': 'docs\\test.epub', 'ch.': '2'}, ...
```

**Pull request closed without merging**